src/forest.rs

```diff
@@ -124,7 +124,10 @@ where
                     return Err(MoreThanOne);
                 }
                 let &split = splits.iter().next().unwrap();
-                let (left, right, _) = node.range.split_at(split);
+                let (left, right) = node
+                    .range
+                    .split_at::<Unknown>(/*split*/ unimplemented!())
```

Comment:
`self.possible_splits` will need to store trusted indices to be able to split here.